templates/mapTpl.go

```diff
@@ -0,0 +1,42 @@
+package templates
+
+import (
+	"github.com/bigbluedisco/protoc-gen-zap/zap"
+	pgs "github.com/lyft/protoc-gen-star"
+)
+
+type mapData struct {
+	MapName   string
+	Getter    string
+	AddFunc   string
+	FormatKey string
+	IsStars   bool
+}
+
+func newMapData(f pgs.Field, obsType zap.ObfuscationType) mapData {
+	formatKey := "key"
+	keyType := f.Type().Key().ProtoType()
+	if isSimple(keyType) && keyType != pgs.StringT {
+		formatKey = "strconv.Itoa(int(key))"
+	}
```

Comment:
This doesn't work for booleans, and I've check the proto docs according to them it can be a map key